packages/dashboard/e2e/tests/utils/dragAndDrop.ts

```diff
@@ -114,27 +115,31 @@ type DragToOptions = {
  * That "jumping" would not cause the right events to fire within the React DnD hooks.
  *
  */
-export const dragAndDrop = (page: Page) => (from: Locator) => ({
-  dragTo: async (to: Locator, options?: DragToOptions) => {
-    const { sourcePosition, targetPosition } = options || {};
-
-    const fromBb = await from.boundingBox();
-    const toBb = await to.boundingBox();
-
-    if (!fromBb || !toBb) throw new Error('could not get dimensions for locators');
-
-    const fromPoint = sourcePosition ? toPointTuple(sourcePosition({ source: fromBb, target: toBb })) : center(fromBb);
-    const toPoint = targetPosition ? toPointTuple(targetPosition({ source: fromBb, target: toBb })) : center(toBb);
-
-    const points = interpolate([fromPoint, toPoint], 2);
-    const [first, ...rest] = points;
-
-    await page.mouse.move(...first);
-    await page.mouse.down({ button: 'left' });
-
-    for (const p of rest) {
-      await page.mouse.move(...p);
-    }
-    await page.mouse.up({ button: 'left' });
-  },
-});
+export const dragAndDrop =
+  (page: Page): DragGenerator =>
+  (from: Locator) => ({
+    dragTo: async (to: Locator, options?: DragToOptions) => {
+      const { sourcePosition, targetPosition } = options || {};
+
+      const fromBb = await from.boundingBox();
+      const toBb = await to.boundingBox();
+
+      if (!fromBb || !toBb) throw new Error('could not get dimensions for locators');
+
+      const fromPoint = sourcePosition
+        ? toPointTuple(sourcePosition({ source: fromBb, target: toBb }))
+        : center(fromBb);
+      const toPoint = targetPosition ? toPointTuple(targetPosition({ source: fromBb, target: toBb })) : center(toBb);
+
+      const points = interpolate([fromPoint, toPoint], 2);
+      const [first, ...rest] = points;
```

Comment:
Consider making these names more specific to avoid required the reader to maintain the meaning of "rest" later in the method.

Reply:
yes, updated